Update lambda-logs-collection-troubleshooting-guide.md (#7858)
DataDog/documentation: content/en/logs/guide/lambda-logs-collection-troubleshooting-guide.md

kind: documentation
---

If you don't see logs forwarded from a Datadog forwarder Lambda function in the Log Explorer, follow the troubleshooting steps below. If you continue to have trouble after following these steps, [contact Datadog support][2] for further assistance.

## Are your logs sent to Datadog?

1. Navigate to the [Log Explorer's Live Tail view][3].
2. In the search bar, use a filter to limit the Live Tail view to just the logs coming from your Lambda function. Some common search queries are:
    * By source: the source is often set to `source:lambda`, `source:aws` or `source:cloudwatch` but you can find other possible sources in the `parse_event_source` function in the [Lambda function][4]. 
    * By forwarder name: the Lambda function adds a `forwardername` tag to all the logs it forwards. You can filter on this tag by searching for `forwardername:*` or `forwardername:<FORWARDER_FUNCTION_NAME>`.

> 3. If you do see the logs in the Live Tail, but not in the Log Explorer, that means your log index has some [exclusion filters][5] setup, they are filtering out your logs.

Reviewer: ```suggestion
3. If you do see the logs in the Live Tail, but not in the Log Explorer, that means your log index has some [exclusion filters][5] set up. These filters are filtering out your logs.
```

### Manage your function triggers

> The forwarder Lambda function needs to have triggers (CloudWatch Logs or S3) set up in order for logs to be forwarded. Follow the steps below to ensure the triggers are set up correctly.

Reviewer: For logs to be forwarded, the forwarder Lambda function needs to have triggers (CloudWatch Logs or S3) set up. Follow the steps below to ensure the triggers are set up correctly.

> 1. Does the source of your log (CloudWatch log group or S3 bucket) shows up in the "Triggers" list in the forwarder Lambda console? If yes, ensure it's enabled. Otherwise follow the steps below to check in the S3 or CloudWatch log group console, because the "Triggers" list displayed in the Lambda console is known to be incomprehensive.

Reviewer: ```suggestion
1. Does the source of your log (CloudWatch log group or S3 bucket) show up in the "Triggers" list in the forwarder Lambda console? If yes, ensure it's enabled. Otherwise, follow the steps below to check in the S3 or CloudWatch log group console, because the "Triggers" list displayed in the Lambda console is known to be incomprehensive.
```

   ```

3. For CloudWatch log group, navigate to the log group's console's "Subscriptions" field under the "Log group details" section. Alternatively, you can make a query using AWS CLI command below. If the log group is not subscribed by the forwarder Lambda function, you need to configure a trigger.
   ```
   aws logs describe-subscription-filters --log-group-name <LOG_GROUP_NAME>
   ```

4. Set triggers [automatically][8] or [manually][9].

Note, AWS doesn't allow for more than one resource to be subscribed to a log source. If your log source is already subscribed by a different resource, you need to remove that subscription first.

For CloudWatch log group, you can use the following metrics within the Datadog platform to confirm whether logs are delivered from the log group to the forwarder Lambda function. Use the `log_group` tag to filter the data when viewing the metrics.

| Metric                          | Description                                                                                        |
|---------------------------------|----------------------------------------------------------------------------------------------------|

> | `aws.logs.incoming_log_events`  | The number of log events uploaded to Cloudwatch Logs                                               |

Reviewer: ```suggestion
| `aws.logs.incoming_log_events`  | The number of log events uploaded to CloudWatch Logs                                               |
```

| `aws.logs.forwarded_log_events` | The number of log events forwarded to the subscription destination                                 |
| `aws.logs.delivery_errors`      | The number of log events failed to be delivered to the subscription destination                    |
| `aws.logs.delivery_throttling`  | The number of log events throttled for delivering to the subscription destination                  |

## Check the Lambda function logs

1. From the monitoring tab, click **View logs in Cloudwatch**.

{{< img src="logs/guide/lambda-logs-cloudwatch.png" alt="Lambda logs in Cloudwatch"  style="width:80%;" >}}

2. Find the most recent log stream.

> 3. Do you see any error? Try search "?ERROR ?Error ?error".

Reviewer: ```suggestion
3. Do you see any errors? Try searching "?ERROR ?Error ?error".
```

> 4. Set environment variable "DD_LOG_LEVEL" to "debug" on the forwarder Lambda function to enable the debugging logs for further debugging. The debugging logs are quite verbose, remember to disable it after debugging.

Reviewer: ```suggestion
4. Set environment variable "DD_LOG_LEVEL" to "debug" on the forwarder Lambda function to enable the debugging logs for further debugging. The debugging logs are quite verbose; remember to disable it after debugging.
```
